clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/BasicAmazonDeployHandler.groovy

```diff
@@ -551,20 +563,21 @@ class BasicAmazonDeployHandler implements DeployHandler<BasicAmazonDeployDescrip
   @PackageScope
   Collection<AmazonBlockDevice> buildBlockDeviceMappings(
     BasicAmazonDeployDescription description,
-    LaunchConfiguration sourceLaunchConfiguration
+    Object sourceLaunchSetting
```

Comment:
This loss of typing and subsequent duck typing of everything in here and in convertBlockDevices isn't that great for our future selfs and their groovy refactoring sanity.

could you make an interface for the things that need do exist on a sourceLaunchSetting (instanceType and List<BlockDeviceMapping> I think is all) and wrap the launchConfiguration/launchTemplate?

Reply:
Agreed! This definitely felt dirty.